File: 560/python/main1.py
# O(n^2)
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        cnt = 0
        for i in range(len(nums)):
            s = 0
            for j in range(i, len(nums)):
                s += nums[j]
                if s == k:
                    cnt += 1
        return cnt

File: 974/rust/main1.rs
impl Solution {
    pub fn subarrays_div_by_k(nums: Vec<i32>, k: i32) -> i32 {
        let mut s = 0;
        let mut m = vec![0; k as usize];
        let mut n = 0;

        m[0] = 1;
        for num in nums {
            s += num;
            let r = ((s % k + k) % k) as usize;
            n += m[r];
            m[r] += 1;
        }
        return n;
    }
}
